illumio_connector.py

```diff
@@ -152,6 +151,286 @@ def _handle_get_traffic_analysis(self, param):
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            ip_list = self._pce.ip_lists.get(
+                policy_version="active", params={"name": name}
+            )
+            for data in ip_list:
+                if name == data.name:
+                    ip_list_data = data
+                    action_result.set_status(
+                        phantom.APP_SUCCESS, "Successfully fetched IP List"
+                    )
+                    break
+            if not ip_list_data:
+                ip_list_data = {}
+                action_result.set_status(
+                    phantom.APP_ERROR,
+                    "IP List with name '{}' does not exist".format(name),
+                )
+
+        except IllumioException as e:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error fetching IP List: {}".format(e),
+            )
+
+        result = self.convert_object_to_json(ip_list_data, action_result)
+        action_result.add_data(result)
+        return action_result.get_status()
+
+    def _handle_create_ruleset(self, param):
+        name = param["name"]
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        rule_set = None
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            ruleset_list = self._pce.rule_sets.get(params={"name": name})
+            for ruleset in ruleset_list:
+                if name == ruleset.name:
+                    rule_set = ruleset
+                    action_result.set_status(
+                        phantom.APP_SUCCESS,
+                        "Found existing ruleset with name {}".format(name),
+                    )
+                    break
+
+            if not rule_set:
+                rule_set = illumio.RuleSet(
+                    name=name, scopes=[illumio.LabelSet(labels=[])]
+                )
+                rule_set = self._pce.rule_sets.create(rule_set)
+                action_result.set_status(
+                    phantom.APP_SUCCESS, "Successfully created ruleset"
+                )
+        except IllumioException as e:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error creating ruleset: {}".format(e),
+            )
+
+        result = self.convert_object_to_json(rule_set, action_result)
+        action_result.add_data(result)
+        return action_result.get_status()
+
+    def handle_comma_seperated_string(self, comma_str):
+        """
+        Convert comma seperated string into list.
+
+        :param comma_str: comma seperated string
+        :return : list
+        """
+        str_to_list = [x.strip() for x in comma_str.split(",")]
+        str_to_list = list(filter(None, str_to_list))
```

Comment:
This `list(filter())` call is unnecessary - you can just add the filter condition to the comprehension above:

```
separated_list = [x.strip() for x in comma_str.split(",") if x]
```

Reply:
Done